R/utils-shared.R

```diff
@@ -277,6 +277,14 @@ isNullOrFalse <- function(variable) {
 splitDataGaps <- function(data, ts, isDV){
   
   data_list <- data[[ts$field[1]]]
+
+  #Add zero/negative gaps
+  if("gaps"  %in% names(data_list)){
```

Comment:
seems like you would want to add the zero negative gaps to the `gaps` field even if it does not exist yet, and that should happen inside of `addZeroNegativeGaps`. Otherwise it should be called `findZeroNegativeGaps`, and then be added as it is doing here. See comment on L114 of `addZeroNegativeGaps`.

If you keep it as `addZeroNegativeGaps`:
```r
 #Add zero/negative gaps
 data_list$gaps <- addZeroNegativeGaps(data_list$gaps, ts$field[1], data, isDV)
```

Otherwise:
```r
#Add zero/negative gaps
zeroNegativeGaps <- findZeroNegativeGaps(ts$field[1], data, isDV)
if("gaps"  %in% names(data_list)){
    data_list$gaps <- rbind(data_list$gaps, zeroNegativeGaps)
} else {
    data_list$gaps <- zeroNegativeGaps
}
```